### PR TITLE
Resolve issue where selected view annotation is not moved to correct z-order

### PR DESCRIPTION
**Pull request by ZiZasaurus**
## Pull request checklist:
 - [ ] Describe the changes in this PR, especially public API changes.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
    
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Add documentation comments for any added or updated public APIs.
 - [ ] Add any new public, top-level symbols to the Jazzy config's `custom_categories` (scripts/doc-generation/.jazzy.yaml)
 - [ ] Add a changelog entry to to bottom of the relevant section (typically the `## main` heading near the top).
 - [ ] Update the guides (internal access only), README.md, and DEVELOPING.md if their contents are impacted by these changes.
 - [ ] If this PR is a `v10.[version]` release branch fix / enhancement, merge it to `main` first and then port to `v10.[version]` release branch.

PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-ios/blob/main/CONTRIBUTING.md#contributor-license-agreement).

### CODE REVIEW

**Review by pjleonard37 — APPROVED** [review]

File: Tests/MapboxMapsTests/Annotations/ViewAnnotationManagerTests.swift [hunk 1]
```diff
@@ -288,6 +288,22 @@ final class ViewAnnotationManagerTests: XCTestCase {
...
 
+    func testAnnotationPlacementZOrder() {
+        let annotationViewA = addTestAnnotationView(id: "test-id")
+        let annotationViewB = addTestAnnotationView(id: "test-id2")
+
+        // Item closest to map should be the last in the annotation array
+        XCTAssertEqual(mapboxMap.addViewAnnotationStub.invocations.last?.parameters.id, "test-id2")
+
+        // After updating, the item clostst to the map is now furthest from the map
+        try? manager.update(annotationViewA, options: ViewAnnotationOptions(selected: true))
```
Comment (maios) [inline Tests/MapboxMapsTests/Annotations/ViewAnnotationManagerTests.swift:299]: this function will not call to `mapboxMap.addViewAnnotation(withId:options:)`, therefore the following assertion does not make sense, the stubbed invocation is made in the previous step of this test case and has not been reset, so, with or without this line the test would always succeed

**Review by maios — CHANGES_REQUESTED** [review]
correct me if I'm wrong but the order of the views is decided in `DelegatingViewAnnotationPositionsUpdateListenerDelegate`, upon receiving changes in view annotations (adding/removing/updating), `mapboxMap` will notify its delegate about new placement of annotation views. I think the test should setup similar to this test below
https://github.com/mapbox/mapbox-maps-ios/blob/6ac39ecdf77ce868c6941036d6a4cc55f0421cfd/Tests/MapboxMapsTests/Annotations/ViewAnnotationManagerTests.swift#L276

File: Tests/MapboxMapsTests/Annotations/ViewAnnotationManagerTests.swift [hunk 2]
```diff
@@ -288,6 +288,22 @@ final class ViewAnnotationManagerTests: XCTestCase {
         XCTAssertEqual(annotationView.frame, CGRect(x: 150.0, y: 200.0, width: 100.0, height: 50.0))
     }
 
+    func testAnnotationPlacementZOrder() {
+        let annotationViewA = addTestAnnotationView(id: "test-id")
+        let annotationViewB = addTestAnnotationView(id: "test-id2")
+
+        // Item closest to map should be the last in the annotation array
+        XCTAssertEqual(mapboxMap.addViewAnnotationStub.invocations.last?.parameters.id, "test-id2")
```
[the remark this answers is not in the document]
Comment (ZiZasaurus) [reply Tests/MapboxMapsTests/Annotations/ViewAnnotationManagerTests.swift]: Looks like bringSubviewToFront/sendToBack doesn't change zPosition, but it does move subview position among the subviews so I referenced the array/ array order instead.

**Review by maios — APPROVED** [review]
Please check if 2 added files in Debug App and Example Apps are intended

**Review by evil159 — APPROVED** [review]

**Pull request merged**